pypeit/extraction.py

```diff
@@ -447,7 +445,7 @@ def run(self, model_noise=None, spat_pix=None):
                 self.slits.mask[reduce_masked], 'BADEXTRACT')
 
         # Return
-        return self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs, self.waveimg, self.tilts
```

Comment:
Is the point of returning self.slits here that they have been modified in the routine? My point is that the slits are an input argument, but seems the bitmask is being modified based on whether the extraction is successful or not, which means the input argument can change.  I think it is indeed better practice to return a new set of slits from this method so that input arguments are immutable, but I just wanted to understand the motivation.

Reply:
That's right... the bitmask is getting updated, so it really should be returned. Technically, `slits` is the input argument, and `self.slits` is modified during extraction. So, I think `self.slits` needs to be returned.

Reply:
Yep I agree. It is better coding practice for arguments that are modified to be immutable, and the modified object to be returned.